pkg/types/types.go

```diff
@@ -76,3 +78,34 @@ type MercuryProvider interface {
 	OnchainConfigCodec() mercury.OnchainConfigCodec
 	ContractTransmitter() mercury.Transmitter
 }
+
+type FunctionsProvider interface {
+	PluginProvider
+	LogPollerWrapper() LogPollerWrapper
+	Sign(data ...[]byte) ([]byte, error)
```

Comment:
We can pass the Keystore served by core in to the Functions ReportingPlugin(Factory) - does that work? Then it can be left out here.

Reply:
That's what we're doing right now. I might've misunderstood your earlier comment. If the plan is to still keep all chain-specific keys in the key store but abstract them behind a chain-agnostic interface then that works too. I'll remove the sign() call from here, thanks.